src/mscorlib/shared/System/Diagnostics/Debug.cs

```diff
@@ -91,18 +91,34 @@ public static void Assert(bool condition, string message, string detailMessage)
             if (!condition)
             {
                 string stackTrace;
-
                 try
                 {
-                    stackTrace = Internal.Runtime.Augments.EnvironmentAugments.StackTrace;
+                    stackTrace = new StackTrace(0, true).ToString(System.Diagnostics.StackTrace.TraceFormat.Normal);
                 }
                 catch
                 {
                     stackTrace = "";
                 }
+                WriteLine(FormatAssert(stackTrace, message, detailMessage));
+                s_ShowDialog(stackTrace, message, detailMessage, "Assertion Failed");
+            }
+        }
 
+        internal static void Assume(bool condition, string message, string detailMessage)
+        {
+            if (!condition)
+            {
+                string stackTrace;
+                try
+                {
+                    stackTrace = new StackTrace(0, true).ToString(System.Diagnostics.StackTrace.TraceFormat.Normal);
+                }
+                catch
+                {
+                    stackTrace = "";
+                }
                 WriteLine(FormatAssert(stackTrace, message, detailMessage));
-                s_ShowAssertDialog(stackTrace, message, detailMessage);
+                s_ShowDialog(stackTrace, message, detailMessage, "Assumption Failed");
```

Comment:
This may rather call `SR.GetResourceString(GetResourceNameForFailure(failureKind))` instead of hard-coding `"Assumption Failed"`.